monitoring/rid_qualifier/aircraft_state_replayer.py

```diff
@@ -0,0 +1,163 @@
+from monitoring.monitorlib.auth import make_auth_adapter
+from monitoring.monitorlib.infrastructure import DSSTestSession
+from monitoring.monitorlib import rid
+import json, os
+import uuid
+from pathlib import Path
+from typing import  Any
+from monitoring.rid_qualifier.utils import OperatorLocation, RIDFlightDetails, TestFlightDetails, TestFlight, TestPayload, DeliverablePayload
+import arrow
+import pathlib
+
+from typing import List
+from monitoring.rid_qualifier.utils import RIDQualifierTestConfiguration
+
+class TestBuilder():
+    ''' A class to setup the test data and create the objects ready to be submitted to the test harness '''
+
+    def __init__(self, test_configuration: RIDQualifierTestConfiguration) -> None:
+        self.test_configuration = test_configuration
+        # Change directory to read the test_definitions folder appropriately
+        p = pathlib.Path(__file__).parent.absolute()
+        os.chdir(p)
+
+        aircraft_states_directory = Path('test_definitions', test_configuration.locale, 'aircraft_states')
+        aircraft_state_files = self.get_aircraft_states(aircraft_states_directory)
+
+        usses = self.test_configuration.usses
+
+        self.disk_rid_state_data =[]
+        for uss_index, uss in enumerate(usses):
+            aircraft_states_path = Path(aircraft_states_directory, aircraft_state_files[uss['allocated_flight_track_number']])
+
+            with open(aircraft_states_path) as generated_rid_state:
+                disk_rid_state_file = json.load(generated_rid_state)
+                self.disk_rid_state_data.append(disk_rid_state_file)
+
+
+    def get_aircraft_states (self, aircraft_states_directory: Path):
+
+        ''' This method checks if there are tracks in the tracks directory '''
+
+        all_files = os.listdir(aircraft_states_directory)
+        files = [f for f in all_files if os.path.isfile(os.path.join(aircraft_states_directory, f))]
+
+        if files:
+            return all_files
```

Comment:
Return even the folders?

Reply:
Thank I have now modified this to send the path in this method.